File: analysis/utils_r/demo_paths.R
parse_demo_mode <- function(args = commandArgs(trailingOnly = TRUE)) {
  "demo" %in% args
}

get_results_dir <- function(demo_mode, ...) {
  file.path(if (demo_mode) "./demo_results" else "./results", ...)
}

get_figures_dir <- function(demo_mode, ...) {
  file.path(if (demo_mode) "./demo_figures" else "./figures", ...)
}

get_results_input_dir <- function(demo_mode, ...) {
  if (demo_mode) {
    demo_path <- get_results_dir(TRUE, ...)
    if (dir.exists(demo_path)) return(demo_path)
  }
  get_results_dir(FALSE, ...)
}

File: analysis/utils_py/demo_paths.py
from __future__ import annotations

import sys
from pathlib import Path


def parse_demo_mode(argv: list[str] | None = None) -> bool:
	args = sys.argv[1:] if argv is None else argv
	return "demo" in args


def get_results_dir(repo_root: Path, *parts: str, demo_mode: bool = False) -> Path:
	base_dir = repo_root / ("demo_results" if demo_mode else "results")
	return base_dir.joinpath(*parts)


def get_results_input_dir(repo_root: Path, *parts: str, demo_mode: bool = False) -> Path:
	if demo_mode:
		demo_path = get_results_dir(repo_root, *parts, demo_mode=True)
		if demo_path.exists():
			return demo_path

	return get_results_dir(repo_root, *parts, demo_mode=False)


def get_figures_dir(repo_root: Path, *parts: str, demo_mode: bool = False) -> Path:
	base_dir = repo_root / ("demo_figures" if demo_mode else "figures")
	return base_dir.joinpath(*parts)